_database-integrations/amazon-s3/amazon-s3-csv-latest.md

```diff
@@ -0,0 +1,630 @@
+              </td>
+              <td>
+                {{ record.magic }}
+              </td>
+            </tr>
+            {% endfor %}
+          </table>
+
+          Before the next job, the CSV file containing these rows is modified. This means that Stitch will replicate the contents of the entire file, including the rows for `Finn` and `Jake` even if they haven't been updated.
+
+          In the destination, the table might now look like the table below. Notice that records for `Finn` and `Jake` have been appended to the end of the table with new `{{ system-column.primary-key }}` values:
+
+          {% assign second-job = integration.sample-data %}
+
+          <table class="attribute-list">
+            <tr>
+              <td width="45%; fixed">
+                <strong>{{ system-column.primary-key }}</strong>
+              </td>
+              <td>
+                <strong>name</strong>
+              </td>
+              <td>
+                <strong>type</strong>
+              </td>
+              <td>
+                <strong>magic</strong>
+              </td>
+            </tr>
+            {% for record in second-job %}
+            <tr>
+              <td>
+                {{ record.primary-key }}
+              </td>
+              <td>
+                {{ record.name }}
+              </td>
+              <td>
+                {{ record.type }}
+              </td>
+              <td>
+                {{ record.magic }}
+              </td>
+            </tr>
+            {% endfor %}
+          </table>
+
+      - title: "Querying Append-Only tables"
+        anchor: "querying-append-only-tables"
+        content: |
+          Querying Append-Only tables requires a different strategy than you might normally use. For instructions and a sample query, check out the [Querying Append-Only tables guide]({{ link.replication.append-only-querying | prepend: site.baseurl }}).
+
+          {% capture append-only-destinations %}
+          The answer depends on your destination:
+
+          - **BigQuery and Amazon S3 CSV** - These destinations are Append-Only regardless of whether Primary Keys are specified.
+
+             If you specified Primary Keys during setup, use the column(s) specified (ex: `id`). Otherwise, use `{{ system-column.primary-key }}`.
+          - **All other destinations** - Use `{{ system-column.primary-key }}` when querying Append-Only tables.
+          {% endcapture %}
+
+          {% include note.html first-line="**What column do I use as a Primary Key when querying?**" content=append-only-destinations %}
+          
+  - title: "Determining data types"
+    anchor: "determining-data-types"
+    content: |
+      To determine a column's data type, Stitch will analyze the first 1,000 lines of (up to) the first **five** files included for a given table.
+
+      Stitch's {{ integration.display_name }} integration will load data from CSV files and type it as one of the following data types:
+
```

Comment:
I like this section. I think it has the correct level of detail. I did want to mention that if a field has mixed types we will type it in a priority.  For instance if you have integers and numbers, we will make it the more generic number.  If it has numbers and strings it will be a string as strings can't be converted to numbers.  That being said I don't think it is necessary detail for the docs.

Reply:
It's an interesting deviation from what we usually do, though. In other integrations, if a column has two data types (like `string` and `boolean`), we'll split the column and append the data type to the second column name (`active` and `active__bool`).